demo/sdl2surface_rawfb/sdl2surface_rawfb.h

```diff
@@ -140,40 +110,49 @@ nk_sdlsurface_line_horizontal(const struct sdlsurface_context *sdlsurface,
      * It does not check for scissors or image borders.
      * The caller has to make sure it does no exceed bounds. */
     unsigned int i, n;
-    unsigned int c[16];
+    unsigned char c[16 * 4];
     unsigned char *pixels = sdlsurface->fb->pixels;
-    unsigned int *ptr;
+    unsigned int bpp = sdlsurface->fb->format->BytesPerPixel;
 
-    pixels += y * sdlsurface->fb->pitch;
-    ptr = (unsigned int *)pixels + x0;
+    pixels += (y * sdlsurface->fb->pitch) + (x0 * bpp);
 
-    n = x1 - x0;
-    for (i = 0; i < sizeof(c) / sizeof(c[0]); i++)
-        c[i] = nk_sdlsurface_color2int(col, sdlsurface->fb->format->format);
+    n = (x1 - x0) * bpp;
+    if (bpp == 4) {
+        for (i = 0; i < sizeof(c) / bpp; i++)
+            ((Uint32 *)c)[i] = nk_sdlsurface_color2int(col, sdlsurface->fb->format);
+    } else if (bpp == 2) {
+        for (i = 0; i < sizeof(c) / bpp; i++)
+            ((Uint16 *)c)[i] = nk_sdlsurface_color2int(col, sdlsurface->fb->format);
+    } else {
+        for (i = 0; i < sizeof(c) / bpp; i++)
+            ((Uint8 *)c)[i] = nk_sdlsurface_color2int(col, sdlsurface->fb->format);
+    }
 
-    while (n > 16) {
-        memcpy((void *)ptr, c, sizeof(c));
-        n -= 16; ptr += 16;
+    while (n > sizeof(c)) {
+        memcpy(pixels, c, sizeof(c));
```

Comment:
I didn't try it but guess that some compilers (MSVC?) might warn about this (maybe in conjunction with C89...).